src/webots/maths/WbRotation.hpp

```diff
@@ -81,7 +81,8 @@ class WbRotation {
   }
 
   // invalid only if |axis| == 0.0
-  bool isValid() const { return !(mX == 0.0 && mY == 0.0 && mZ == 0.0); }
+  // the mAngle == mAngle checks the NaN case.
+  bool isValid() const { return !(mX == 0.0 && mY == 0.0 && mZ == 0.0) && (mAngle == mAngle); }
```

Comment:
Interesting, I use to use `isnan(mAngle)` instead, but it is fine that way (and requires no dependency on math.h)

Reply:
yes I saw that after having committed the code that is was used in other places when I did the review.
Do you want me to replace it?

Reply:
I believe it is indeed slightly more readable (will save us the comment).